doc/cephfs/cache-configuration.rst

```diff
@@ -209,3 +209,62 @@ cache. The limit is configured via:
 
 It is not recommended to set this value above 5M but it may be helpful with
 some workloads.
+
+
+Dealing with "clients failing to respond to cache pressure" messages
+--------------------------------------------------------------------
+
+Every second (or every interval set by the ``mds_cache_trim_interval``
+configuration paramater), the MDS runs the "cache trim" procedure. One of the
+steps of this procedure is "recall client state". During this step, the MDS
+checks every client (session) to determine whether it needs to recall caps.
+If any of the following are true, then the MDS needs to recall caps:
+
+1. the cache is full (the ``mds_cache_memory_limit`` has been exceeded) and
+   needs some inodes to be released
+2. the client exceeds ``mds_max_caps_per_client`` (1M by default)
+3. the client is inactive
+
+To determine whether a client (a session) is inactive, the session's
+``cache_liveness`` parameters is checked and compared with the value::
+
+   (num_caps >> mds_session_cache_liveness_magnitude)
+
+where ``mds_session_cache_liveness_magnitude`` is a config param (``10`` by
+default). If ``cache_liveness`` is smaller than this calculated value, the
+session is considered inactive and the MDS sends a "recall caps" request for
+all cached caps (the actual recall value is ``num_caps -
+mds_min_caps_per_client(100)``).
+
+If the client does not release the caps fast enough, the MDS repeats the
+"recall caps" request one second later. This means that the MDS will send
+"recall caps" with high value again and again. The "total" counter of "recall
+caps" for the session will grow and will eventually exceed the mon warning limit.
+There is a throttling mechanism, controlled by mds_recall_max_decay_threshold
```

Comment:
From line 243 to line 270, the English language could stand a grammar and elegance pass. I want to make certain that I don't misrepresent the technical information in those lines, so I will be making this grammar-and-elegance pass through the medium of GitHub, with the collaboration of the CephFS team.